### ISSUE TITLE
GDPR compliance.

### ISSUE THREAD

**Issue by hgeorgsch** [KQMATH/moodle-mod_capquiz#76]
Employ the Privacy document and assess privacy compliance. Address the issue in README/doc's.

**Comment by skrede** [comment]
I'm not familiar with GDPR other than it exists and govern privacy, but the little I've picked up, shouldn't this be an issue within Moodle itself? 

The only information about the user that Capquiz accumulate is answers for questions and an overall rating on how well they are answered. It's not using any cookies, except the ones Moodle is using without Capquiz being aware.

The information Capquiz gathers, question results and rating, is available in the database, and this might be further processed and used by the Moodle host / provider, it is really the service provider that must be compliant. 

Looking at the bigger picture, it would be cumbersome for every plugin to implement a dialog gathering permissions / acknowledgements on how the information is used. 

Quizzes and questionaires are at the core of Moodle, and as such, and I think it should be Moodle itself that must provide compliance. 

TL;DR 
https://expresswriters.com/tldr-what-is-gdpr/

**Comment by andstor** [comment]
Moodle solves most of the GDPR compliancy within the Moodle core itself. However, all 3rd party plugins have to implement the [**Privacy API**](https://docs.moodle.org/dev/Privacy_API). It is therefore up to the plugin developer(s) to implement this API correctly. This also applies if the plugin doesn't store any user data. When the implementation is done correctly, Moodle provides the necessary tools for ensuring GDPR compliance. This is mainly the ability to read all personal stored user data, and also being able to delete it.

**Comment by danmarsden** [comment]
Please note - Moodle's core unit tests fail when a plugin is installed that does not comply with the privacy api so organisations that rely on Continuous Integration to deploy their site will also not be able to use your plugin until this is fixed.

This is not currently a blocker for approval in the plugins db - but in 6 months time this policy may change and if you have not managed to get your plugin approved by then, you may be required to implement it prior to approval.

**Comment by sebastsg** [comment]
This should be solved now. Tested with the export/delete scripts provided by Moodle docs, and also via the frontend by making requests. 

No unit tests are made yet, however.
